src/Symfony/Component/Intl/DateFormatter/DateFormat/FullTransformer.php

```diff
@@ -105,7 +105,7 @@ public function format(\DateTime $dateTime)
      *
      * @throws NotImplementedException When it encounters a not implemented date character
      */
-    public function formatReplace($dateChars, $dateTime)
+    public function formatReplace(string $dateChars, \DateTime $dateTime)
```

Comment:
Did this method work with `DateTimeImmutable` instances before your change? In that case, we should type-hint `DateTimeInterface` instead.

Reply:
Doesn't seems so, `$dateTime` is only called with:
https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Intl/DateFormatter/DateFormat/Transformer.php#L32

In the following logic.

Reply:
That seems correct according to the current implementation. But I've realized this is buggy: #32898

Reply:
Nice catch @Tobion, I'll wait for #32898 to be resolved and I'll rebase on it with a `\DateTimeInterface` type-hint then

Reply:
we can't wait on #32898 as it's not clear when it's fixed unless you work on it yourself. but for now it's fine to merge it like this.

Reply:
This type-hint is not dependant on what `IntlDateFormatter::format` accepts. `IntlDateFormatter::format` takes the timestamp of a `DateTime` instance, and creates a new instance of `DateTime` which is then passed to the transformer.  So I think this can safely be updated to `DateTimeInterface` without the changes to #32898

Reply:
@pierredup you're right.
I suggest to remove all the internal `@param \DateTime` as it does not provide value